controlplane/src/core/services/AccessTokenAuthenticator.ts

```diff
@@ -0,0 +1,43 @@
+import AuthUtils from '../auth-utils.js';
+import { OrganizationRepository } from '../repositories/OrganizationRepository.js';
+
+export type AccessTokenAuthContext = {
+  userId: string;
+  organizationId: string;
+};
+
+export default class AccessTokenAuthenticator {
+  constructor(private orgRepo: OrganizationRepository, private authUtils: AuthUtils) {}
+
+  /**
+   * Authenticates the user with the given access token. Returns the user's organization ID and user's ID.
+   *
+   * @param accessToken
+   */
+  public async authenticate(accessToken: string): Promise<AccessTokenAuthContext> {
+    const userInfoData = await this.authUtils.getUserInfo(accessToken);
+
+    // TODO will have to change when wesupport multiple orgs
+    const orgSlug = userInfoData.groups[0].split('/')[1];
+
+    const organization = await this.orgRepo.bySlug(orgSlug);
+
+    if (!organization || !organization?.id) {
+      throw new Error('Organization deosnt exist');
```

Comment:
typo and throw Unauthorized error here so we don't log this as an error.

Reply:
Done